src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,202 @@
+        BookstoreOwner bookstoreOwner = bookstoreOwnerRepository.findById(bookstoreOwnerID);
+        if (bookstoreOwner == null)
+            return null;
+        Bookstore bookstore = new Bookstore(name);
+        bookstore.setBookstoreOwner(bookstoreOwner);
+        bookstoreOwner.addBookstore(bookstore);
+        bookstoreRepository.save(bookstore);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstore;
+    }
+
+
+    //Book REST endpoints
+    @GetMapping("/api/getBook")
+    public Book getBook(@RequestParam(value = "id") long id) {
+        return bookRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBooks")
+    public Iterable<Book> getBooks() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/api/getBooksByBookstore")
+    public Iterable<Book> getBooksByBookstore(@RequestParam(value = "bookstoreID") long bookstoreID) {
+        Bookstore bookstore = bookstoreRepository.findById(bookstoreID);
+        if (bookstore == null)
+            return null;
+        return bookRepository.findByBookstore(bookstore);
+    }
+
+    @PostMapping("/api/newBook")
+    public Book newBook(@RequestParam(value = "name") String name, @RequestParam(value = "isbn") String isbn, @RequestParam(value = "picture") String picture, @RequestParam(value = "description") String description, @RequestParam(value = "author") String author, @RequestParam(value = "publisher") String publisher, @RequestParam(value = "bookstoreID") long bookstoreID) {
+        Bookstore bookstore = bookstoreRepository.findById(bookstoreID);
+        if (bookstore == null)
+            return null;
+        Book book = new Book(name, isbn, picture, description, author, publisher);
+        bookstore.addBook(book);
+        book.setBookstore(bookstore);
+        bookRepository.save(book);
+        bookstoreRepository.save(bookstore);
+        return book;
+    }
+
+
+    //Customer REST endpoints
+    @GetMapping("/api/getCustomer")
+    public Customer getCustomer(@RequestParam(value = "id") long id) {
+        return customerRepository.findById(id);
+    }
+
+    @GetMapping("/api/getCustomers")
+    public Iterable<Customer> getCustomers() {
+        return customerRepository.findAll();
+    }
+
+    @PostMapping("/api/newCustomer")
+    public Customer newCustomer(@RequestParam(value = "name") String name, @RequestParam(value = "address") String address, @RequestParam(value = "email") String email, @RequestParam(value = "phoneNumber") String phoneNumber) {
+        Customer customer = new Customer(name, address, email, phoneNumber);
+        customerRepository.save(customer);
+        return customer;
+    }
+
+
+    //Shopping Cart REST endpoints
+    @GetMapping("/api/getShoppingCart")
+    public ShoppingCart getShoppingCart(@RequestParam(value = "id") long id) {
+        return shoppingCartRepository.findById(id);
+    }
+
+    @GetMapping("/api/getShoppingCarts")
```

Comment:
Wonder whether we want to expose this via REST?

Reply:
Addressed in 0f27f09dc1598aeea21d1460bc54bc99cf5a5a3f